### DISCUSSION TITLE
Ubuntu Server autostart.

### DISCUSSION THREAD

**Discussion by bjoern-gringer** [mtrossbach/noah-mqtt#43] [Q&A]
First: **Thank you for the great tool!**

My Linux knowledge is medium.
The tool started by hand, is working perfekt but I have a problem with the autostart of noah-mqtt.

This is what I have done:
`sudo nano /etc/systemd/system/noah-mqtt.service`

I filled the file with:

`________________________________________________________________
[Unit]
Description=Noah-mqtt wird gestartet
After=network.target

[Service]
Environment=NODE_ENV=production
Type=notify
ExecStart=GROWATT_USERNAME=growatt_XXXXX GROWATT_PASSWORD=XXXXXXXXXX3MTH33 MQTT_HOST=192.168.178.46 MQTT_PORT=1883 ./noah-mqtt
WorkingDirectory=/opt/noah-mqtt
StandardOutput=inherit
StandardError=inherit
WatchdogSec=10s
Restart=always
RestartSec=10s
User=bjoern

[Install]
WantedBy=multi-user.target
_________________________________________________________________
`
Next:

`sudo systemctl daemon-reload`

`sudo systemctl enable noah-mqtt`

`sudo systemctl start noah-mqtt`

But now, if I check the status with:

`sudo systemctl status noah-mqtt`

I get the Failuremessage:

_________________________________________________________________
`

`Nov 26 16:17:11 bjoern-GK41 systemd[1]: Starting Noah-mqtt wird gestartet...

Nov 26 16:17:11 bjoern-GK41 systemd[7997]: noah-mqtt.service: Failed to locate executable GROWATT_USERNAME=growatt_XXXX: No such file or directory

Nov 26 16:17:11 bjoern-GK41 systemd[7997]: noah-mqtt.service: Failed at step EXEC spawning GROWATT_USERNAME=growatt_XXX: No such file or directory

Nov 26 16:17:11 bjoern-GK41 systemd[1]: noah-mqtt.service: Main process exited, code=exited, status=203/EXEC

Nov 26 16:17:11 bjoern-GK41 systemd[1]: noah-mqtt.service: Failed with result 'exit-code'.

Nov 26 16:17:11 bjoern-GK41 systemd[1]: Failed to start Noah-mqtt wird gestartet.
_________________________________________________________________
`
I hope someone can help me.

Nice greetings from Hamburg Björn

**Comment by xdatravelbug** [comment]
I am using this great add-on within FHEM on a Raspi.
Here is what I did and was working after some trial and error:

1. Make noah-mqtt executable (at least I had this issue on my setup causing the main headache)
chmod 750 /opt/fhem/noah-mqtt/noah-mqtt

2. Create service:
sudo nano /etc/systemd/system/noah-mqtt.service

3. Add the following (and modify as needed)

[Unit]
Description=MQTT service for NOAH2000
Wants=network-online.target
After=network.target

[Service]
Type=simple
User=root
#Environment=LOG_LEVEL=info
#Environment=GROWATT_SERVER_URL=
Environment=GROWATT_USERNAME=YOUR_NAME
Environment=GROWATT_PASSWORD=YOUR_PW
Environment=MQTT_HOST=localhost
Environment=MQTT_PORT=1884
#Environment=MQTT_CLIENT_ID=noah-mqtt
#Environment=MQTT_USERNAME=
#Environment=MQTT_PASSWORD=
#Environment=MQTT_TOPIC_PREFIX=noah2mqtt
Environment=POLLING_INTERVAL=180
Environment=BATTERY_DETAILS_POLLING_INTERVAL=180
Environment=PARAMETER_POLLING_INTERVAL=180
ExecStart=/opt/fhem/noah-mqtt/noah-mqtt
Restart=always

[Install]
WantedBy=multi-user.target

4. Exit and save

5. Reload systemd manager configuration
sudo systemctl daemon-reload

6. Start service
sudo systemctl start noah-mqtt.service

7. Activate service
sudo systemctl enable noah-mqtt.service

8. Check service
sudo systemctl status noah-mqtt.service

Best regards

**Reply by bjoern-gringer** [reply]
Great thanks, 
with your tutorial it works perfekt.

Gretting from an other fhem user ;-)
